ext/openssl/xp_ssl.c

```diff
@@ -1912,7 +1912,7 @@ static int php_openssl_enable_crypto(php_stream *stream,
 		}
 
 		timeout = sslsock->is_client ? &sslsock->connect_timeout : &sslsock->s.timeout;
-		has_timeout = !sslsock->s.is_blocked && (timeout->tv_sec || timeout->tv_usec);
+		has_timeout = !sslsock->s.is_blocked && (timeout->tv_sec > 0 || (timeout->tv_sec == 0 && timeout->tv_usec));
```

Comment:
This looks late to me. We should avoid creating a negative timeout in the first place (and use a zero timeout instead).

Reply:
Hmm, a zero timeout currently always causes failure. Our `poll()` fallback does it similar to this solution:
https://github.com/php/php-src/blob/12c59f6660706321f9d42c55421ff6864439c8b7/main/network.c#L1233-L1243
And the [`poll(2)` man page](https://man7.org/linux/man-pages/man2/poll.2.html) states:
> Specifying a negative value in timeout means an infinite timeout.